### PR TITLE
Update Intro section and add concepts

### PR DESCRIPTION
**Pull request by srinjoyc**
This PR aims to refresh the content on the gentle introduction page by separating out conceptual information to a new concepts section. While the themes of the content and general structure was preserved, several edits are made to improve readability, update key items based on new information, and align better with the developer journey.

**Updates:**
- In a nutshell section revised to 3 points reflective of top value props of stylus
- Use cases are more detailed and aligned to value props, with a focus on efficient on-chain verification, DeFi instruments, and high-performance on-chain logic (reflects what most builders are using Stylus for and would be effective in doing so), linking to a specific [case study](https://blog.arbitrum.io/renegade-stylus-case-study/).
-  Offers a step-by-step guide on getting started with direct links to resources like the quickstart, examples, tutorials, and community channels (Telegram, Discord, GitHub). **Note: Some of these will need to be updated before merging to master (relies on other content section like tutorials existing).**
- Contains more hyperlinks to resources
- Actively seeks developers for building and contributing, emphasizing the $5M grant funding through the Stylus Sprint.
- Updates some language

**Removes**
- Public Preview banner from the gentle introduction page
- Conceptual content on 'how Stylus works' to a new page

### CODE REVIEW

File: arbitrum-docs/stylus/stylus-gentle-introduction.md [hunk 1]
```diff
@@ -9,123 +9,34 @@ sidebar_position: 1
...
+- <b>Advanced DeFi Instruments</b>:  Power complex financial instruments and processes like synthetic assets, options, and futures with onchain computation via extending current protocols (ie. Uniswap V4 hooks) or building your own.
+- <b>High-Performance On-Chain Logic</b>: Support memory and compute-intensive applications like onchain games and generative art either by writing all of the application in Stylus or enhance performance of existing Solidity contracts by optimizing specific parts.
 
-To keep up with the latest announcements and updates about Stylus:
+### Getting Started
+1. Utilize our [quickstart](https://docs.arbitrum.io/stylus/stylus-quickstart), [examples](https://docs.arbitrum.io/stylus/stylus-quickstart), and [tutorials](https://docs.arbitrum.io/stylus/stylus-quickstart) to help you start building.
+2. Join our Stylus Developer [Telegram](t.me/arbitrum_stylus) group and [Arbitrum Discord](https://discord.gg/arbitrum) for support as well as the official Arbitrum ([@Arbitrum](https://twitter.com/arbitrum)) and Arbitrum Developers ([@ArbitrumDevs](https://twitter.com/ArbitrumDevs)) X accounts for announcements.
```
Comment (hkalodner) [inline arbitrum-docs/stylus/stylus-gentle-introduction.md:38]: Absolutely links like this need to be like https://t.me/arbitrum_stylus

Comment (hkalodner) [reply arbitrum-docs/stylus/stylus-gentle-introduction.md:38]: That's why the vercel preview failed

Comment (srinjoyc) [reply arbitrum-docs/stylus/stylus-gentle-introduction.md]: ty! fixed

File: arbitrum-docs/stylus/stylus-gentle-introduction.md [hunk 2]
```diff
@@ -9,123 +9,34 @@ sidebar_position: 1
...
-
 ### In a nutshell:
 
-- Stylus lets you write smart contracts in programming languages that compile to WASM, such as **Rust, C, C++, and many others**.
-- Rich language support already exists for Rust: you can use the Stylus SDK and CLI tool to start building today.
-- Stylus smart contracts benefit from **Arbitrum's EVM equivalence** thanks to a second, coequal WASM virtual machine.
-- Solidity contracts and Stylus contracts are fully interoperable. In Solidity, you can call a Rust program and vice versa.
-- Stylus contracts are significantly faster and have lower gas fees due to the superior efficiency of <a data-quicklook-from="wasm">WASM</a> programs.
-- Stylus makes it viable to consume RAM on the blockchain because it can greatly optimize memory use, enabling new use cases.
+- Stylus lets you write smart contracts in programming languages that compile to WASM, such as **Rust, C, C++, and many others** allowing you to tap into their ecosystem of libraries and tools. Rich language and tooling support already exists for Rust, try out the SDK and CLI with the [quickstart](https://docs.arbitrum.io/stylus/stylus-quickstart) today.
```
Comment (anegg0) [inline arbitrum-docs/stylus/stylus-gentle-introduction.md:14]:
```suggestion
- Stylus lets you write smart contracts in programming languages that compile to WASM, such as **Rust, C, C++, and many others** allowing you to tap into their ecosystem of libraries and tools. Rich language and tooling support already exists for Rust, try out the SDK and CLI with the [quickstart](https://docs.arbitrum.io/stylus/stylus-quickstart) today.
```
```suggestion
- Stylus lets you write smart contracts in programming languages that compile to WASM, such as **Rust, C, C++, and many others**, allowing you to tap into their ecosystem of libraries and tools. Rich language and tooling support already exists for Rust. You can try the SDK and CLI with the [quickstart](https://docs.arbitrum.io/stylus/stylus-quickstart).
```

Comment (anegg0) [reply arbitrum-docs/stylus/stylus-gentle-introduction.md:14]: It sounds probably better with "soft" CTAs. OCL's docs tone is more declarative than promotional.

File: arbitrum-docs/stylus/stylus-gentle-introduction.md [hunk 3]
```diff
@@ -9,123 +9,34 @@ sidebar_position: 1
...
+With a WASM VM, any programming language that can compile to WASM is within Stylus's scope. While many popular programming languages can be compiled into WASM, some compilers are more suitable for smart contract development than others, like Rust, C, and C++. Other languages like Go, Sway, Move, and Cairo can also be supported. Languages that include their own runtimes, like Python and Javascript, are more complex for Stylus to support, although not impossible. Compared to using Solidity, WASM programs are much more efficient for memory intensive applications. There are many reasons for this, including the decades of compiler development for Rust and C. WASM also has a faster runtime than the EVM, resulting in faster execution. Third-party [contribution](#contributing) in the form of libraries for new and existing languages is welcomed!
```
Comment (anegg0) [inline arbitrum-docs/stylus/stylus-gentle-introduction.md:26]:
```suggestion
With a WASM VM, any programming language that can compile to WASM is within Stylus's scope. While many popular programming languages can be compiled into WASM, some compilers are more suitable for smart contract development than others, like Rust, C, and C++. Other languages like Go, Sway, Move, and Cairo can also be supported. Languages that include their own runtimes, like Python and Javascript, are more complex for Stylus to support, although not impossible. Compared to Solidity, WASM programs are much more efficient for memory-intensive applications. There are many reasons for this, including the decades of compiler development for Rust and C. WASM also has a faster runtime than the EVM, resulting in faster execution. Third-party [contribution](#contributing) in the form of libraries for new and existing languages is welcomed!
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+---
+title: How Stylus Works
+sidebar_label: How Stylus Works
+---
```
Comment (anegg0) [inline arbitrum-docs/stylus/concepts/how-it-works.md:4]:
```suggestion
---
id: stylus-how-it-works
title: 'How Stylus works'
sidebar_label: 'How Stylus works'
description: 'Learn what powers Stylus'
author: srinjoyc
SME: srinjoyc
user_story: As an Ethereum developer/project owner, I need to vet the Stylus.
content_type: concept
---
```

**Review by anegg0 — APPROVED** [review]
This overhaul helps put the right info in front of the right users, I think it's a great job.
I added a few nits, but LGTM

**Pull request merged**